src/libs/kata-types/src/config/hypervisor/mod.rs

```diff
@@ -0,0 +1,1061 @@
+    ///
+    /// If set, when GUEST_PANICKED event occurred, guest memory will be dumped to host filesystem
+    /// under guest_memory_dump_path. This directory will be created automatically if it does not
+    /// exist. The dumped file(also called vmcore) can be processed with crash or gdb.
+    ///
+    /// # WARNING:
+    ///  Dump guest’s memory can take very long depending on the amount of guest memory and use
+    /// much disk space.
+    #[serde(default)]
+    pub guest_memory_dump_path: String,
+}
+
+impl DebugInfo {
+    /// Adjust the configuration information after loading from configuration file.
+    pub fn adjust_configuration(&mut self) -> Result<()> {
+        Ok(())
+    }
+
+    /// Validate the configuration information.
+    pub fn validate(&self) -> Result<()> {
+        Ok(())
+    }
+}
+
+/// Virtual machine device configuration information.
+#[derive(Debug, Default, Deserialize, Serialize)]
+pub struct DeviceInfo {
+    /// Bridges can be used to hot plug devices.
+    ///
+    /// Limitations:
+    /// - Currently only pci bridges are supported
+    /// - Until 30 devices per bridge can be hot plugged.
+    /// - Until 5 PCI bridges can be cold plugged per VM.
+    ///
+    /// This limitation could be a bug in qemu or in the kernel
+    /// Default number of bridges per SB/VM:
+    /// - unspecified or 0   --> will be set to @DEFBRIDGES@
+    /// - > 1 <= 5           --> will be set to the specified number
+    /// - > 5                --> will be set to 5
+    #[serde(default)]
+    pub default_bridges: u32,
+
+    /// VFIO devices are hotplugged on a bridge by default.
+    ///
+    /// Enable hotplugging on root bus. This may be required for devices with a large PCI bar,
+    /// as this is a current limitation with hotplugging on a bridge.
+    #[serde(default)]
+    pub hotplug_vfio_on_root_bus: bool,
+
+    /// Before hot plugging a PCIe device, you need to add a pcie_root_port device.
+    ///
+    /// Use this parameter when using some large PCI bar devices, such as Nvidia GPU.
+    /// The value means the number of pcie_root_port.
+    /// This value is valid when hotplug_vfio_on_root_bus is true and machine_type is "q35"
+    #[serde(default)]
+    pub pcie_root_port: u32,
+
+    /// Enable vIOMMU, default false
+    ///
+    /// Enabling this will result in the VM having a vIOMMU device. This will also add the
+    /// following options to the kernel's command line: intel_iommu=on,iommu=pt
+    #[serde(default)]
+    pub enable_iommu: bool,
+
+    /// Enable IOMMU_PLATFORM, default false
+    ///
+    /// Enabling this will result in the VM device having iommu_platform=on set
+    #[serde(default)]
+    pub enable_iommu_platform: bool,
+}
+
+impl DeviceInfo {
+    /// Adjust the configuration information after loading from configuration file.
+    pub fn adjust_configuration(&mut self) -> Result<()> {
+        if self.default_bridges > 5 {
+            self.default_bridges = 5;
```

Comment:
Please use a constant (and again the question about silent failures)

Reply:
1. use constant to replace 5
2. we can silently adjust the config info